### ISSUE TITLE
cmd/go: go test returns "ld: warning: PIE disabled." on OSX

### ISSUE THREAD

**Issue by arekkas** [golang/go#21918]
### What version of Go are you using (`go version`)?

```
go version go1.9 darwin/amd64
```

### Does this issue reproduce with the latest release?

Yes, before Go 1.9 this problem did not exist.

### What operating system and processor architecture are you using (`go env`)?

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/aeneas/workspaces/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/c3/sys_sm612kg6mph4r640m_bc0000gn/T/go-build948664642=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

### What did you do?

Since installing Go 1.9 (from source w/ gvm, brew, and binary from page) I get a warning when running `go test`. You should be able to reproduce this with:

```
go get github.com/ory/hydra
cd $GOPATH/src/github.com/ory/hydra
glide install
go test -short $(glide novendor)
```
### What did you expect to see?

The test process running without warnings.

### What did you see instead?

```
ld: warning: PIE disabled. Absolute addressing (perhaps -mdynamic-no-pic) not allowed in code signed PIE, but used in type..eqfunc.[106]string from /var/folders/c3/sys_sm612kg6mph4r640m_bc0000gn/T/go-link-904935252/go.o. To fix this warning, don't compile with -mdynamic-no-pic or link with -Wl,-no_pie
```

**Comment by arekkas** [comment]
Also noteworthy, since Go 1.9 on Windows 10 64bit my Antivirus (Avira) detects a virus when running `go test -short $(glide novendor)` on the same project.

**Comment by ianlancetaylor** [comment]
Can you show us the output of `go test -x -short $(glide novendor)`?

**Comment by hirochachacha** [comment]
Dup of https://github.com/golang/go/issues/21776. 
https://github.com/ory/hydra/blob/4651a2303a695ed79955ae64ec9fd7e5dc697906/config/backend_plugin.go#L30

**Comment by arekkas** [comment]
Thanks!

**Comment by rsc** [comment]
Duplicate so removing milestone.

**Comment by rsc** [comment]
Fixed by 88a1e85
